kbase-extension/static/kbase/js/common/jobManager.js

```diff
@@ -299,41 +266,223 @@ define(['common/html', 'common/jobs', 'common/ui', 'util/string'], (html, Jobs,
                 }
             });
             this.model.setItem('exec.jobs', jobIndex);
-            this.runUpdateHandlers(jobArray);
+            this.runHandler('modelUpdate', jobArray);
             return this.model;
         }
 
-        initBatchJob(message) {
-            const { parent_job_id, child_job_ids } = message;
+        /* UTIL FUNCTIONS */
 
-            if (
-                !parent_job_id ||
-                !child_job_ids ||
-                Object.prototype.toString.call(child_job_ids) !== '[object Array]' ||
-                !child_job_ids.length
-            ) {
-                throw new Error(
-                    'Batch job must have a parent job ID and at least one child job ID'
-                );
+        /**
+         * Get the IDs of jobs with a certain status
+         *
+         * @param {array} statusList - array of statuses to find
+         * @param {array} validStates - array of valid statuses for this action (optional)
+         * @returns {array} job IDs to perform the action upon
+         */
+        getJobIDsByStatus(statusList, validStates) {
+            if (validStates && validStates.length) {
+                const allInTheList = statusList.every((status) => {
+                    return validStates.includes(status);
+                });
+                if (!allInTheList) {
+                    console.error(
+                        `Invalid status supplied! Valid statuses: ${validStates.join(
+                            '; '
+                        )}; supplied: ${statusList.join('; ')}.`
+                    );
+                    return null;
+                }
             }
 
-            // initialise `exec.jobs` with the new child jobs
-            this.model.setItem(
-                'exec.jobs',
-                Jobs.jobArrayToIndexedObject(
-                    child_job_ids.map((id) => {
-                        return { job_id: id, status: 'created', created: 0 };
-                    })
-                )
-            );
-            this.model.setItem('exec.jobState', parent_job_id);
-        }
-
-        getFsmStateFromJobs() {
-            const jobsByStatus = this.model.getItem(`exec.jobs.byStatus`);
-            return Jobs.getFsmStateFromJobs(jobsByStatus);
+            const jobsByStatus = this.model.getItem('exec.jobs.byStatus');
+            if (!jobsByStatus || !Object.keys(jobsByStatus).length) {
+                return null;
+            }
+            return statusList
+                .map((status) => {
+                    return jobsByStatus[status] ? Object.keys(jobsByStatus[status]) : [];
+                })
+                .flat();
         }
     }
 
+    const jobActionsMixin = (Base) =>
```

Comment:
Why the mixin use here? Just for code compartmentalization? Are there variants of the JobManagerCore that won't use these functions?

Reply:
Alternately, can this mixin be used elsewhere?

Reply:
Code compartmentalisation, yes, and also I'd like to be able to create a version for the single app cell, which would have different functions enabled and different ways to init the job (see the BatchInitMixin).